activerecord/lib/active_record/log_subscriber.rb

```diff
@@ -20,18 +20,14 @@ def initialize
       @odd = false
     end
 
-    def render_bind(attribute)
-      value = if attribute.type.binary? && attribute.value
-        if attribute.value.is_a?(Hash)
-          "<#{attribute.value_for_database.to_s.bytesize} bytes of binary data>"
-        else
-          "<#{attribute.value.bytesize} bytes of binary data>"
-        end
+    def render_bind(attr, type_casted_value)
+      value = if attr.type.binary? && attr.value
+        "<#{attr.value_for_database.to_s.bytesize} bytes of binary data>"
```

Comment:
I realized a issue. We cannot render legacy binds even if master branch.
If passed legacy binds, `attr` is an array, cannot call `attr.type.binary?`.